Semikolon in Whitelist + Add new boolean property: BtchBookg

MicrotronX opened the pull request:
plus unittests

MicrotronX committed:
Add ; (semikolon) to whitelist to be able to use escape in xml

MicrotronX committed:
Add new boolean property: BtchBookg

MicrotronX committed:
Add new boolean property: BtchBookg

aspettl commented on SEPACreditTransfer.pas:
```diff
@@ -83,6 +83,7 @@   TCreditTransferTransactionInformation = class
   TCreditTransferPaymentInformation = class
   private
     fPmtInfId: String;                                 // payment information identification
+    fBtchBookg:boolean;                                // Batchbooking parameter (at Sepa level default true if not defined)
```
Please format the same as the other lines (whitespace before type, type with capital letter) - also in the other file, of course.

aspettl commented on SEPACreditTransfer.pas:
```diff
@@ -271,6 +273,8 @@ constructor TCreditTransferPaymentInformation.Create;
   fDbtrAcct         := TAccountIdentification.Create;
   fDbtrAgt          := TFinancialInstitution.Create;
+
+  fBtchBookg    :=  false; // Set it default to false, we want to have details
```
I do not agree on the default. According to the documentation, "true" is the default and for details one needs a special agreement. We cannot assume here that this is what the user has and wants.

aspettl commented on SEPACreditTransfer.pas:
```diff
@@ -373,11 +377,18 @@ function TCreditTransferPaymentInformation.Validate(const schema: String; const
+    function _Boolean2Xml(_value:boolean):string;
+    begin
+        if _value then result:='true' else result:='false';
+    end;
```
This function is defined twice in this PR. Please extract to the "SEPACommon.pas" file (and adapt indentation and whitespaces to the used code style, please also remove the "_" prefix in the variable name).

aspettl commented on SEPACommon.pas:
```diff
@@ -245,6 +245,9 @@ function CharIsSEPAWhitelisted(const c: Char): Boolean;
             (c = ',') or (c = '-') or (c = ' ') or
             (c = '(') or (c = '+') or (c = '.') or
+
+            (c = ';') or   // We allow the ; char so we can use XML-Escapes like &amp;
```
I'm still not sure this is the right way. On the other hand, it gives the flexibility to use escaping when the own financial institution accepts this.
In the future: please prepare separate branches/pull requests for different features.

MicrotronX replied:
We don't have problems with xml-escapes with customers in germany, austria and luxembourg. At the moment this is working without problems. Without escaping all customers had problems with special chars.

aspettl commented on README.md:
```diff
@@ -1,4 +1,4 @@
-# Delphi-Unit zur Erstellung von SEPA-XML-Dateien
+# Delphi-Unit zur Erstellung von SEPA-XML - Dateien
```
Please revert, I don't see any reason for this change.

aspettl reviewed (changes requested):
Thanks for the pull request! I have added a few comments, would be great if you can implement these changes.

MicrotronX committed:
- Moved _Boolean2XML to SEPACommon as SEPABoolean2Xml
- Changed boolean to Boolean with whitespace before
- Reverted README.md to original

MicrotronX commented:
Hello Aaron,

wir haben soeben alle deine vorgeschlagenen Änderungen durchgeführt. Die
Readme war nur ein Test auf unserer Seite ob es überhaupt mit dem Fork /
Pullrequest funktioniert (ist das erste mal ;-)
Werde gleich einen neuen Pull-Request erstellen.

Mfg. Yusuf Zorlu

2018-08-04 13:47 GMT+02:00 Aaron Spettl <notifications@github.com>:

> *@aspettl* requested changes on this pull request.
>
> Thanks for the pull request! I have added a few comments, would be great
> if you can implement these changes.
> ------------------------------
>
> In SEPACreditTransfer.pas
> <https://github.com/aspettl/delphi-sepa-xml/pull/4#discussion_r207706686>:
>
> > @@ -83,6 +83,7 @@   TCreditTransferTransactionInformation = class
>    TCreditTransferPaymentInformation = class
>    private
>      fPmtInfId: String;                                 // payment information identification
> +    fBtchBookg:boolean;                                // Batchbooking parameter (at Sepa level default true if not defined)
>
> Please format the same as the other lines (whitespace before type, type
> with capital letter) - also in the other file, of course.
> ------------------------------
>
> In SEPACreditTransfer.pas
> <https://github.com/aspettl/delphi-sepa-xml/pull/4#discussion_r207706721>:
>
> > @@ -271,6 +273,8 @@ constructor TCreditTransferPaymentInformation.Create;
>    fChrgBr           := CHRG_BR_SLEV;
>    fDbtrAcct         := TAccountIdentification.Create;
>    fDbtrAgt          := TFinancialInstitution.Create;
> +
> +  fBtchBookg    :=  false; // Set it default to false, we want to have details
>
> I do not agree on the default. According to the documentation, "true" is
> the default and for details one needs a special agreement. We cannot assume
> here that this is what the user has and wants.
> ------------------------------
>
> In SEPACreditTransfer.pas
> <https://github.com/aspettl/delphi-sepa-xml/pull/4#discussion_r207706753>:
>
> > @@ -373,11 +377,18 @@ function TCreditTransferPaymentInformation.Validate(const schema: String; const
>  procedure TCreditTransferPaymentInformation.SaveToStream(const stream: TStream; const schema: String);
>  var
>    i: Integer;
> +    function _Boolean2Xml(_value:boolean):string;
> +    begin
> +        if _value then result:='true' else result:='false';
> +    end;
>
> This function is defined twice in this PR. Please extract to the
> "SEPACommon.pas" file (and adapt indentation and whitespaces to the used
> code style, please also remove the "_" prefix in the variable name).
> ------------------------------
>
> In SEPACommon.pas
> <https://github.com/aspettl/delphi-sepa-xml/pull/4#discussion_r207706798>:
>
> > @@ -245,6 +245,9 @@ function CharIsSEPAWhitelisted(const c: Char): Boolean;
>              (c = '''') or (c = ':') or (c = '?') or
>              (c = ',') or (c = '-') or (c = ' ') or
>              (c = '(') or (c = '+') or (c = '.') or
> +
> +            (c = ';') or   // We allow the ; char so we can use XML-Escapes like &amp;
>
> I'm still not sure this is the right way. On the other hand, it gives the
> flexibility to use escaping when the own financial institution accepts this.
> In the future: please prepare separate branches/pull requests for
> different features.
> ------------------------------
>
> In README.md
> <https://github.com/aspettl/delphi-sepa-xml/pull/4#discussion_r207706828>:
>
> > @@ -1,4 +1,4 @@
> -# Delphi-Unit zur Erstellung von SEPA-XML-Dateien
> +# Delphi-Unit zur Erstellung von SEPA-XML - Dateien
>
> Please revert, I don't see any reason for this change.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/aspettl/delphi-sepa-xml/pull/4#pullrequestreview-143380574>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ARO4let4D3B3Sp7MfbgoG2W7S4MidgLMks5uNYnLgaJpZM4VtkHg>
> .
>

MicrotronX committed:
formatting changes

aspettl commented:
Danke!
